### PR TITLE
Bugfix Constructed hammer

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
None

#### Purpose of change
Cause constructed hammer to vanish with use of spell.

#### Testing
Untested will attempt to do so this weekend.

### CODE REVIEW

File: data/mods/Xedra_Evolved/spells/dreamer_spells.json [hunk 1]
```diff
@@ -393,8 +393,26 @@
...
+  },
+  {
+    "id": "EOC_DESTRUCT_ITEM_IN_HANDS",
+    "type": "effect_on_condition",
+    "name": "Constructed Hammer Self destruct",
+    "description": "Removes constructed hammer.  You can't see it except in debug mode.",
+    "effect": [
+      { "u_remove_item_with": "constructed_hammer" }
+    ]
+  }
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/spells/dreamer_spells.json:415]:
```suggestion
  },
```

**Comment by akrieger** [comment]
Maybe name the eoc something specific to the hammer being destroyed?

**Comment by Maleclypse** [comment]
> Maybe name the eoc something specific to the hammer being destroyed?

Originally when I found it in docs as an example I thought it would be an EOC to destroy whatever the pc was wielding. I wasn’t able to figure out how to make it do that so I should change the name.

File: data/mods/Xedra_Evolved/spells/dreamer_spells.json [hunk 2]
```diff
@@ -393,7 +393,23 @@
...
+  {
+    "id": "constructed_hammer_self_destruct",
+    "type": "SPELL",
+    "name": "Constructed Hammer Self destruct",
+    "description": "Removes constructed hammer.  You can't see it except in debug mode.",
+    "effect": "effect_on_condition",
+    "effect_str": "EOC_DESTRUCT_ITEM_IN_HANDS"
+  },
+  {
+    "id": "EOC_DESTRUCT_ITEM_IN_HANDS",
```
Comment (GuardianDll) [inline data/mods/Xedra_Evolved/spells/dreamer_spells.json:408]:
```suggestion
    "effect_str": "EOC_REMOVE_CONSTRUCTED_HAMMER"
  },
  {
    "id": "EOC_REMOVE_CONSTRUCTED_HAMMER",
```

File: data/mods/Xedra_Evolved/spells/dreamer_spells.json [hunk 3]
```diff
@@ -393,7 +393,23 @@
...
     "max_damage": 500,
     "min_range": 1,
-    "max_range": 1
+    "max_range": 1,
+    "extra_effects": [ { "id": "constructed_hammer_self_destruct" } ]
+  },
+  {
+    "id": "constructed_hammer_self_destruct",
+    "type": "SPELL",
+    "name": "Constructed Hammer Self destruct",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/spells/dreamer_spells.json:402]:
```suggestion
    "shape": "blast",
    "name": "Constructed Hammer Self destruct",
```

File: data/mods/Xedra_Evolved/spells/dreamer_spells.json [hunk 4]
```diff
@@ -393,7 +393,23 @@
...
+    "description": "Removes constructed hammer.  You can't see it except in debug mode.",
+    "effect": "effect_on_condition",
+    "effect_str": "EOC_REMOVE_CONSTRUCTED_HAMMER"
+  },
+  {
+    "id": "EOC_REMOVE_CONSTRUCTED_HAMMER",
+    "type": "effect_on_condition",
+    "name": "Constructed Hammer Self destruct",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/spells/dreamer_spells.json:410]:
```suggestion
    "shape": "blast",
    "name": "Constructed Hammer Self destruct",
```

**Comment by GuardianDll** [comment]
resolves #66129

**Pull request merged**